### DISCUSSION TITLE
unetr_btcv_segmentation tutorial output labeling class error

### DISCUSSION THREAD

**Discussion by jykkim111** [Project-MONAI/tutorials#681] [Q&A]
Hi, 

I was just trying out the unetr_btcv_segmentation tutorial, but with a smaller scope. 
The only change I made from the tutorial is changing the labels so that they have 5 classes instead of 14 [0, 1, 2, 3, 4] including the background. Yes, I did change the model out_channels parameter to 5 along with all the parameters.

![image](https://user-images.githubusercontent.com/32403499/166405906-4b8868bb-37d4-42f4-bd1e-dacb4e2eaf57.png)

![image](https://user-images.githubusercontent.com/32403499/166405978-70f4900c-0c16-490e-944e-5d356ba9142c.png)

Just for a sanity check, I used 1 CT image to see if it overfits and it does overfit well, but when I check out the visualizations of the output, the class labels turn out all wrong. I am having trouble locating the problem. Its weird because the dice scores come out properly, but it shouldn't if the output class labels are incorrect.

![image](https://user-images.githubusercontent.com/32403499/166406042-4b6c9fd7-9519-41da-b4b4-1cda117a4525.png)

In the input label the labels are: 
0 - background, 
1 - gallbladder, 
2 - liver, 
3 - stomach, 
4 - pancreas

In the output label, the labels are:
0 - background, 
1 - liver, 
2 - stomach, 
3 - nothing, missing**
4 - pancreas

![image](https://user-images.githubusercontent.com/32403499/166405483-da0663b7-3d55-4423-8ed6-03dc7fbba6b3.png)

**Comment by ahatamiz** [comment]
Hi @jykkim111 

It is not straightforward to pinpoint the problem just from these descriptions. However would be great to try : (1) printing out per class dice values to see if all classes converge properly (2) try including one or two more samples to see if this problem exists. we should be able to overfit to those as well. 

Thanks

**Reply by jykkim111** [reply]
Thank you for the quick reply @ahatamiz 

I've tried to follow the per class dice values to see if all classes converge properly during the validation 
while training.

![image](https://user-images.githubusercontent.com/32403499/166429121-a3b0905e-cc43-48a1-ad93-4e4926008c2c.png)

![image](https://user-images.githubusercontent.com/32403499/166429161-b5020a43-ae50-4833-95e0-c81417c77005.png)

It seems like there is something wrong with the training as class 1 should be the gallbladder which I am guessing should be the slowest to converge. I've been looking at the DiceMetric source code, but I'm not sure what triggers the Nan value and why the classes get mixed up.
